### PR TITLE
dhall: fix warnings about cabal sandboxes

### PR DESCRIPTION
**Pull request by pthariensflame**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Resolves `dhall`'s part of #55253.

### CODE REVIEW

File: Formula/dhall.rb [hunk 1]
```diff
@@ -16,10 +17,11 @@ class Dhall < Formula
   end
 
   depends_on "cabal-install" => :build
-  depends_on "ghc@8.6" => :build
+  depends_on "ghc" => :build
```
Comment (Bo98) [inline Formula/dhall.rb:20]: :+1:

File: Formula/dhall.rb [hunk 2]
```diff
@@ -16,10 +17,11 @@ class Dhall < Formula
...
 
   def install
-    install_cabal_package
+    system Formula["cabal-install"].bin/"cabal", "v2-update"
+    system Formula["cabal-install"].bin/"cabal", "v2-install", *std_cabal_v2_args
```
Comment (Bo98) [inline Formula/dhall.rb:24]: Is there a situation `cabal` would not be in the superenv PATH or there would be more than one (and thus require the absolute path done here)?

**Review by Bo98 — COMMENTED** [review]
Please remove the `require "language/haskell"` and the `include Language::Haskell::Cabal` too.

File: Formula/dhall.rb (hunk 2 above)
Comment (pthariensflame) [reply Formula/dhall.rb:24]: Not that I can think of.  I was coding defensively but I'll ease up.

**Review by Bo98 — APPROVED** [review]
Thanks!

**Pull request closed without merging**